rpc/gnmi/gnmi-specification.md

```diff
@@ -1109,18 +1138,23 @@ array), the following considerations apply:
 
 *   In the case that multiple attribute values are required to uniquely address
     an element - e.g., `/a/f[k1=10][k2=20] `- and a replace or update
-    operation's path specifies  a subset of the attributes (e.g., `/a/f[k1=10]`)
-    then this MUST be considered an error by the target system - and an status
-    code of` InvalidArgument (3)` specified.
+    operation's path specifies a strict subset of the attributes (e.g.,
+    `/a/f[k1=10]`, `/a/f`), then this MUST be considered an error by the target
```

Comment:
This change means that one cannot send `/path/to/list` with a body that is an array of objects in JSON (assuming JSON_IETF) right? Is there a reason to disallow this?

Reply:
I didn't change anything here -- a strict/proper subset is likely what the original sentence meant to say.

I think there is no clear reason to disallow this, since the keys for each element in the array should be provided in the update message as well. However, that would now mean this spec needs to be changed, right?

Reply:
I think there's an ambiguity here, I think that this was intended to say a non-null subset - i.e., keys were partially specified.

I think there are a few cases here:

1. No keys are specified - e.g., `/interfaces/interface` - in this case, I think I expect that the serialised content of the list would be specified. e.g., the array of objects for JSON_IETF.
2. A partial set of keys are specified - this is an error with `InvalidArgument` in my view.

I think it'd be worth checking implementations to see whether the no-keys variant is supported. This was the original motivation for having surrounding containers in OpenConfig such that a get + set could be done on the surrounding container to get all entries.